addons/paint-snap/state.js

```diff
@@ -0,0 +1,67 @@
+export let snapOn = true;
+
+const ls = localStorage;
+
+export const snapTo = {
+  pageEdges: true,
+  pageCenter: true,
+  pageCorners: false,
+  objectEdges: true,
+  objectCenters: true,
+  objectCorners: false,
+};
+export const snapFrom = {
+  boxCenter: true,
+  boxCorners: false,
+  boxEdgeCenters: false,
+};
+
+export let threshold = 10;
+
+export let guideColor;
+
+if (ls.getItem("sa-paint-snap-snapTo")) {
+  const stored = JSON.parse(ls.getItem("sa-paint-snap-snapTo"));
+  Object.assign(snapTo, stored);
+}
+
+if (ls.getItem("sa-paint-snap-snapFrom")) {
+  const stored = JSON.parse(ls.getItem("sa-paint-snap-snapFrom"));
+  Object.assign(snapFrom, stored);
+}
+
+if (ls.getItem("sa-paint-snap-threshold")) {
+  const stored = JSON.parse(ls.getItem("sa-paint-snap-threshold"));
+  threshold = stored;
+}
+
+export function setSnapTo(point, to) {
+  snapTo[point] = !!to;
+  ls.setItem("sa-paint-snap-snapTo", JSON.stringify(snapTo));
+}
+export function setSnapFrom(point, to) {
+  snapFrom[point] = !!to;
+  ls.setItem("sa-paint-snap-snapFrom", JSON.stringify(snapFrom));
+}
+
+export function setThreshold(thresh) {
+  threshold = thresh;
+  ls.setItem("sa-paint-snap-threshold", JSON.stringify(threshold));
+}
```

Comment:
This is inconsistent with onion skinning. @WorldLanguages should we add tons of settings like in onion skinning? Or should we have onion skinning do this? Or maybe we add an API to handle storage?

Reply:
I vote api.

Reply:
API was created (in #2843) but it wasn't merged...

Reply:
Ummm

![image](https://user-images.githubusercontent.com/70162741/190242296-d7a74069-d817-4d0f-bcdc-929b52b5cd6a.png)

Reply:
Related: #1950

Reply:
somewhat, yes

Reply:
I don't think it makes sense to delay the release of this addon until we have a good way to handle cases like this.

I'd say we make this addon handle settings just like onion-skinning for the initial release of this addon.